docs/api/structures/web-request-filter.md

```diff
@@ -1,4 +1,6 @@
 # WebRequestFilter Object
 
-* `urls` string[] - Array of [URL patterns](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Match_patterns) that will be used to filter out the requests that do not match the URL patterns.
-* `types` String[] (optional) - Array of types that will be used to filter out the requests that do not match the types. When not specified, all types will be matched. Can be `mainFrame`, `subFrame`, `stylesheet`, `script`, `image`, `font`, `object`, `xhr`, `ping`, `cspReport`, `media` or `webSocket`.
+* `urls` string[] (optional) _Deprecated_ - Array of [URL patterns](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Match_patterns) used to include requests that match these patterns. An empty array matches all URLs.
```

Comment:
I completely agree with you on matching to Chrome's standards!

We could just change the behavior and make it a breaking change, along with `types`. 
My concern is that `urls` is a required field so it will definitely break the behavior for a small group of ppl who may have wanted to filter all the urls that match a certain type as an example. On the other hand, `types` is an optional field so I would feel fine about changing the behavior because I don't see ppl adding an empty array unless they have a particular type they want to filter for. 

The tradeoff is that we have a more usable and standard API in the long term and we can make clear release notes to point out the breaking change!

Reply:
> My concern is that urls is a required field so it will definitely break the behavior for a small group of ppl who may have wanted to filter all the urls that match a certain type as an example.

This is a valid concern. One approach we could take would be to detect usage of `urls: []` and log a deprecation warning that this behavior will be doing away and to recommend using `urls: ['<all_urls>']`. After a deprecation period, we move to the fixed behavior.

-----

I wanted to check this usage so I [expanded my search](https://github.com/search?q=%2F%28%5C%29%7Csession%7Cses%29%5C.webRequest.*urls%2F&type=code)

Active projects using `urls: []`
- https://github.com/pocketnetteam/pocketnet.gui/blob/ed7021d4f619f4333009b8553738ad8527813104/tpls/main.js.tpl#L626
- https://github.com/vladimiry/ElectronMail/blob/5cc4b92a45bf65be2f789a04e2f89b511e193885/src/electron-main/web-request/index.ts#L119

Active projects using `urls: ['<all_urls>']`
- https://github.com/ghostery/adblocker/blob/master/packages/adblocker-electron/src/index.ts#L96
- https://github.com/Milkshiift/GoofCord/blob/fa9855e8d0873b7f0f6f4909db591ce7541d4155/src/modules/firewall.ts#L33

Reply:
Thanks for going above an beyond to do the search @samuelmaddock ! 

I'll do as you suggested!
- Detect usage of `urls: []` and modify it to  `['<all_urls>']`. Which I'm currently doing in this PR
- Log deprecation message if we detect the use of empty url
- Don't add the new includeUrls 

The only conflict will be if someone actually wants to use empty array to include nothing rather than include everything. But I can't think of an instance when a user would want to add a filter and not include any urls. 😅

Also, are you ok with me deleting the logic to match `types: []` as matching all types? I think we should keep the logic consistent that empty array matches nothing. 

Used your search and don't see many usages of types so we might be able to do this cleanly. I can still log a deprecating message. 

https://github.com/search?q=%2F%28%5C%29%7Csession%7Cses%29%5C.webRequest.*types%2F&type=code